prow/jenkins/jenkins.go

```diff
@@ -221,7 +231,18 @@ func (c *Client) doRequest(method, path string) (*http.Response, error) {
 // the ProwJob is going to be used as the buildId parameter that will help
 // us track the build before it's scheduled by Jenkins.
 func (c *Client) Build(pj *kube.ProwJob) error {
-	return c.BuildFromSpec(&pj.Spec, pj.Metadata.Name)
+	start := time.Now()
```

Comment:
With the logging proposal we will also want to log all interactions with the Jenkins client. Does it seem reasonable to have the Jenkins client we use here decorated twice (with a metrics endpoint and with a logger) so that these actions are opaque in the controller?

Reply:
All jenkins actions are already logged. This method used to have a c.log call but it has moved inside BuildFromSpec.

Reply:
We want to inject the logger though -- to add context like which ProwJob we are making the call for, etc.

Reply:
We already print the type of the job, the name of the job, and the name of the prowjob. What's missing is the event-GUID from Github. We could store it as an annotation in the ProwJob and print it if it exists.

Reply:
Hm, not sure -- seems like the separation of concerns that we get from allowing to pass sub-loggers with more context added is nicer. The log line in the client just prints the things that are important to the client and by virtue of inheriting the larger context for the logger we get more information. When we add another thing we want to track, who wants to go through every Jenkins call and edit the log line to add in that field?

Reply:
Yeah, not opposed to that. We already log 90% of the stuff we need today is all.

Reply:
Or 75% more precisely. lol